webrtc/src/rtp_transceiver/srtp_writer_future.rs

```diff
@@ -12,6 +12,70 @@ use interceptor::{Attributes, RTCPReader, RTPWriter};
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, Weak};
 use tokio::sync::Mutex;
+use util;
+
+/// RTP packet sequence number manager.
+pub(crate) struct SequenceTransformer(util::sync::Mutex<SequenceTransformerInner>);
+
+struct SequenceTransformerInner {
+    offset: u16,
+    last_sq: u16,
+    reset_needed: bool,
+    enabled: bool,
+    data_sent: bool,
+}
+
+impl SequenceTransformer {
+    pub(crate) fn new() -> Self {
+        Self(util::sync::Mutex::new(SequenceTransformerInner {
+            offset: 0,
+            last_sq: rand::random(),
+            reset_needed: false,
+            enabled: false,
+            data_sent: false,
+        }))
+    }
+
+    pub(crate) fn enable(&self) -> Result<()> {
+        let mut guard = self.0.lock();
+
+        if guard.enabled {
+            return Err(Error::ErrRTPSenderSeqTransEnabled);
+        }
+
+        (!guard.data_sent)
+            .then(|| {
+                guard.enabled = true;
+            })
+            .ok_or(Error::ErrRTPSenderDataSent)
+    }
+
+    pub(crate) fn reset_offset(&self) {
+        self.0.lock().reset_needed = true;
+    }
+
+    fn seq_number(&self, raw_sn: u16) -> Option<u16> {
+        let mut guard = self.0.lock();
+
+        if !guard.enabled {
+            return None;
+        }
+
+        let offset = guard
+            .reset_needed
+            .then(|| {
+                guard.reset_needed = false;
+                let offset = guard.last_sq.overflowing_sub(raw_sn.overflowing_sub(1).0).0;
+                guard.offset = offset;
+                offset
```

Comment:
we don't need to reset guard.last_sq here, right?

Reply:
Right, here we just getting or calculating `offset`.